velox/exec/tests/AggregationFuzzer.cpp

```diff
@@ -1016,40 +1064,105 @@ void AggregationFuzzer::verifyAggregation(
           "Velox and DuckDB results don't match");
     }
 
-    std::vector<core::PlanNodePtr> plans;
-    makeAlternativePlans(
-        groupingKeys, aggregates, masks, projections, input, plans);
+    testPlans(plans, verifyResults, resultOrError);
 
-    // Evaluate same plans on flat inputs.
-    std::vector<RowVectorPtr> flatInput;
-    for (const auto& vector : input) {
-      auto flat = BaseVector::create<RowVector>(
-          vector->type(), vector->size(), vector->pool());
-      flat->copy(vector.get(), 0, 0, vector->size());
-      flatInput.push_back(flat);
+  } catch (...) {
+    if (!reproPersistPath_.empty()) {
+      persistReproInfo(plans, reproPersistPath_);
     }
+    throw;
+  }
+}
 
-    makeAlternativePlans(
-        groupingKeys, aggregates, masks, projections, flatInput, plans);
+void AggregationFuzzer::verifyAggregation(
+    std::vector<core::PlanNodePtr> plans) {
+  VELOX_CHECK_GT(plans.size(), 0);
+  auto plan = plans.front();
+  auto node = dynamic_cast<const core::AggregationNode*>(plan.get());
+  auto projectionNode = dynamic_cast<const core::ProjectNode*>(plan.get());
+
+  if (!node) {
+    // This code is tied to plan created by previous verifyAggregation function
```

Comment:
nit: how about making this a top level comment in this function and mentioning: 
It expects plans previously created by the other verifyAggregation() as follows: Projection(optional) <- Aggregation <- Values. Changes there will require corresponding changes here.